### PR TITLE
Add watch tasks for linting and testing

### PR DESCRIPTION
**Pull request by rserur**
Run `yarn watch` to run `yarn lint:watch` and `yarn test:watch` at the same time during development.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,14 +7,18 @@
     "node": ">=8.10.0"
   },
   "scripts": {
-    "test": "./node_modules/.bin/mocha --reporter spec",
+    "docs": "./node_modules/.bin/jsdoc --configure jsdoc.json",
     "lint": "./node_modules/.bin/eslint **/*.js --ignore-pattern node_modules/",
-    "docs": "./node_modules/.bin/jsdoc --configure jsdoc.json"
+    "lint:watch": "chokidar --initial './lib' './test' './examples' -c 'yarn lint'",
```
Comment (bkaney) [inline package.json]: do we need `./node_modules/.bin/chokidar` ?

Comment (rserur) [reply package.json]: Here we're technically using `chokidar-cli`, which already implicitly pulls in `chokidar` as a dependency.

Comment (bkaney) [reply package.json]: I think the pattern we want to follow is for libraries from `package.json` it is safer to reference any cli/bins as `./node_modules/.bin/xxx`.

Comment (bkaney) [reply package.json]: (yarn would be an exception, as it's recommended to always be installed globally)

Comment (rserur) [reply package.json]: Updated to use `./node_modules/.bin`

**Pull request merged**